### PR TITLE
Incorporate Cypress support

### PR DESCRIPTION
**Pull request by rexagod**
A few notable changes:
* Cypress ESLint configurations.
* Cypress meta and reporter configurations.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,14 +3,38 @@
     "baseUrl": ".",
```
Comment (rexagod) [inline tsconfig.json:3]: I sorted this file so as to reduce merge conflicts in the future, but let me know if you want to preserve the original order.

**Comment by rexagod** [comment]
Need to sync to clear up a few concerns.

File: package.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
     "lint": "yarn eslint --ext .ts,.tsx ./src/ && yarn lint-css",
     "lint-fix": "yarn eslint --fix --ext .ts,.tsx ./src/ && yarn lint-css-fix",
     "prepare": "husky install",
+    "test-cypress": "node_modules/.bin/cypress open --config-file ./integration-tests-cypress/cypress-ceph.json --env openshift=true",
     "i18n": "i18next \"src/**/*.{js,ts,tsx}\" [-oc] -c i18next-parser.config.js"
   },
   "dependencies": {
```
Comment (GowthamShanmugam) [inline package.json:25]: "pluralize": "^8.0.0"

**Comment by rexagod** [comment]
Will push helper methods (odf-console's custom `cy.install`, `cy.login`, etc.) in a follow-up PR as discussed in the sync-up.

File: cypress/plugin.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        rules: [
+          {
+            test: /\.tsx?$/,
+            loader: 'ts-loader',
+            options: { happyPackMode: true, transpileOnly: true },
+          },
+        ],
+      },
+    },
+  };
```
Comment (rexagod) [inline cypress/plugin.js:23]: Not sure about this, can we remove this? It seems to look for `tsx` files inside cypress (where it won't find any).

**Pull request merged**